File: Tutorial2InjectConstructor/src/main/java/com/example/tutorial2injectconstructor/di/SharedPreferencesModule.java
package com.example.tutorial2injectconstructor.di;

import android.content.Context;
import android.content.SharedPreferences;

import com.example.tutorial2injectconstructor.model.MySharedPreferences;

import javax.inject.Named;
import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
public class SharedPreferencesModule {

    private Context context;

    public SharedPreferencesModule(Context context) {
        this.context = context;
    }

    @Provides
    @Singleton
    SharedPreferences provideSharedPreferences() {
        return context.getSharedPreferences("PrefName", Context.MODE_PRIVATE);
    }
}

File: Tutorial2-1InjectConstructor/src/main/java/com/example/tutorial2injectconstructor/di/SharedPreferencesModule.kt
package com.example.tutorial2injectconstructor.di

import android.content.Context
import android.content.SharedPreferences
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module
class SharedPreferencesModule(private val context: Context) {

    @Provides
    @Singleton
    fun provideSharedPreferences(): SharedPreferences {
        return context.getSharedPreferences("PrefName", Context.MODE_PRIVATE)
    }

}